enterprise/views.py

```diff
@@ -375,32 +390,122 @@ def get(self, request):
         course_id = request.GET.get('course_id', '')
         program_uuid = request.GET.get('program_uuid', '')
         self.preview_mode = bool(request.GET.get('preview_mode', False))
-        if not (enterprise_customer_uuid and success_url and failure_url):
-            raise Http404
+
+        # Get enterprise_customer to start in case we need to render a custom 404 page
+        # Then go through other business logic to determine (and potentially overwrite) the enterprise customer
+        enterprise_customer = get_enterprise_customer_or_404(enterprise_customer_uuid)
```

Comment:
should i be catching the http404 everywhere we call get_enterprise_customer_or_404?

Reply:
part of the reason why i left the old http404 from get_enterprise_customer_or_404 is because, if we don't have the enterprise customer, then we don't have what we need to get the proper context to render the page, and therefore are unable to properly render the custom error page